### PR TITLE
docs - generate config file

### PR DESCRIPTION
**Pull request by jeff-bruemmer**
This PR adds a new command that generates a config file template. The template is embedded in a Markdown file for documentation.

You can generate the config file template by changing into the top-level Metabase directory and running:

```
clojure -M:doc:ee config-template
```

The bulk of the code is dedicated to a) grabbing all of the `defsettings` available to configure in the config file and b) including the settings' default values. The code then includes some sample database and user settings as examples (supplied by a hand-written yaml file).

This PR reuses some code used to generate the environment variable docs.

## Feedback requested

- Are all the relevant settings included in the config template?
- Any settings we should exclude?
- Anything else we should test?

## Potential next steps

It would be nice to be able to configure settings in the UI then export a config file with the settings and their current values, maybe with an option to export only those settings that have been changed (i.e., where the values differ from the default values). We could reuse some of the code here, though we'd need to include database and user settings in that export.

### CODE REVIEW

File: src/metabase/public_settings.clj [hunk 1]
```diff
@@ -563,6 +563,7 @@ See [fonts](../configuring-metabase/fonts.md).")
   :visibility :public
   :type       :string
   :audit      :getter
+  :default   "https://www.metabase.com/help/premium"
```
Comment (jeff-bruemmer) [inline src/metabase/public_settings.clj:566]: Is this right?

File: enterprise/backend/src/metabase_enterprise/sso/integrations/sso_settings.clj [hunk 2]
```diff
@@ -105,19 +105,16 @@ on your IdP, this usually looks something like `http://www.example.com/141xkex60
 
 (defsetting saml-attribute-email
   (deferred-tru "SAML attribute for the user''s email address")
-  :default "http://schemas.xmlsoap.org/ws/2005/05/identity/claims/emailaddress"
```
Comment (jeff-bruemmer) [inline enterprise/backend/src/metabase_enterprise/sso/integrations/sso_settings.clj:108]: Removed these defaults as they caused issues. Should we keep them and filter out in the code?

File: src/metabase/public_settings.clj (hunk 1 above)
Comment (jeff-bruemmer) [inline src/metabase/public_settings.clj:566]:
```suggestion
  :default   "https://www.metabase.com/help"
```

File: src/metabase/public_settings.clj [hunk 3]
```diff
@@ -563,6 +563,7 @@ See [fonts](../configuring-metabase/fonts.md).")
   :visibility :public
   :type       :string
   :audit      :getter
+  :default   "https://www.metabase.com/help"
```
Comment (jeff-bruemmer) [inline src/metabase/public_settings.clj:566]: Is this correct?

**Review by camsaul — APPROVED** [review]

**Pull request merged**